### PR TITLE
fix dotnet build warnings and code analysis warnings

### PR DESCRIPTION
**Pull request by kuojianlu**
Resolves https://github.com/microsoft/teams-ai/issues/417.

### CODE REVIEW

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/OpenAI/OpenAIClient.cs [hunk 1]
```diff
@@ -8,7 +8,9 @@
 using Microsoft.TeamsAI.Exceptions;
 using Microsoft.TeamsAI.Utilities;
 
+#pragma warning disable IDE0130 // Namespace does not match folder structure
 namespace Microsoft.TeamsAI.OpenAI
```
Comment (swatDong) [inline dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/OpenAI/OpenAIClient.cs:12]: How about just using `Microsoft.TeamsAI.AI.OpenAI`?

Comment (kuojianlu) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/OpenAI/OpenAIClient.cs]: Just realized that changing this namespace will not affect samples. Updated.

**Review by swatDong — APPROVED** [review]

**Comment by corinagum** [comment]
@kuojianlu could you add a summary in the description explaining the reasons for these changes? :)

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/AI.cs [hunk 2]
```diff
@@ -22,21 +22,20 @@ namespace Microsoft.TeamsAI.AI
...
     {
         private readonly IActionCollection<TState> _actions;
-        private readonly AIOptions<TState> _options;
 
         public AI(AIOptions<TState> options, ILogger? logger = null)
         {
             Verify.ParamNotNull(options);
 
-            _options = options;
+            Options = options;
```
Comment (corinagum) [inline dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/AI.cs:30]: What's the reason for this rename?

**Review by corinagum — APPROVED** [review]
I left a few requests, but after that, LGTM!

**Review by singhk97 — APPROVED** [review]

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/AI.cs (hunk 2 above)
Comment (kuojianlu) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/AI.cs:30]: To fix the warning [IDE0032](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0032).

**Comment by kuojianlu** [comment]
> @kuojianlu could you add a summary in the description explaining the reasons for these changes? :)

@corinagum All the changes are to fix the building warnings and code analysis warnings.
image

**Pull request merged**